frontend/app/src/components/UserNoteForm.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <v-form v-model="valid" class="pt-2">
+    <div>
+      <v-text-field
+        :value="value.title"
+        outlined
+        :placeholder="$t('notes_menu.labels.title')"
+        @input="input({ title: $event })"
+      />
+    </div>
+    <div>
+      <v-textarea
+        :value="value.content"
+        outlined
+        :rules="contentRules"
+        :placeholder="$t('notes_menu.labels.content')"
+        @input="input({ content: $event })"
+      />
+    </div>
+  </v-form>
+</template>
+<script lang="ts">
+import {
+  defineComponent,
+  PropType,
+  ref,
+  toRefs,
+  watch
+} from '@vue/composition-api';
+import { get } from '@vueuse/core';
+import { UserNote } from '@/types/notes';
+
+export default defineComponent({
+  name: 'UserNoteForm',
+  props: {
+    value: {
+      required: true,
+      type: Object as PropType<Partial<UserNote>>
+    }
+  },
+  emits: ['input', 'valid'],
+  setup(props, { emit }) {
+    const { value } = toRefs(props);
+
+    const valid = ref(false);
+
+    watch(valid, value => emit('valid', value));
+
+    const input = (newInput: Partial<UserNote>) => {
+      emit('input', { ...get(value), ...newInput });
+    };
+
+    return {
+      valid,
+      input
+    };
+  },
+  data() {
+    return {
+      contentRules: [
+        (v: string) =>
+          !!v || this.$t('notes_menu.rules.content.non_empty').toString()
```

Comment:
In newer components can we use `vuelidate` for validation instead?
For examples, you can check the new settings components.